### PR TITLE
Added Swap Nodes In Pairs of Linkedlist problem on Leetcode

### PR DESCRIPTION
**Pull request by anupamahalder**

### CODE REVIEW

File: Leetcode/24/Swap_Nodes_In_Pairs.cpp [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      ListNode *next;
+      ListNode() : val(0), next(nullptr) {}
+      ListNode(int x) : val(x), next(nullptr) {}
+      ListNode(int x, ListNode *next) : val(x), next(next) {}
+};
+
+class Solution {
+public:
+    ListNode* swapPairs(ListNode* head) {
+        if(head == NULL || head->next == NULL) return head;
```
Comment (nitishhsinghhh) [inline Leetcode/24/Swap_Nodes_In_Pairs.cpp:21]: ```if (head || head->next) return head;```

C++ way of doing the same and is suffice.

File: Leetcode/24/Swap_Nodes_In_Pairs.cpp [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+          
+        // Initialize pointers
+        ListNode *newHead = head->next;
+        ListNode *tmp = head;
+        ListNode *prev = NULL;
+      
+        while(tmp != NULL && tmp->next != NULL){
```
Comment (nitishhsinghhh) [inline Leetcode/24/Swap_Nodes_In_Pairs.cpp:28]: tmp && tmp-> next would help here. 

We can rename the variables for better clarity and use 'current' and 'previous' for 'tmp' and 'prev', respectively.

File: Leetcode/24/Swap_Nodes_In_Pairs.cpp [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            ListNode *next_node = tmp->next;
+
+            // store the next node of the next node 
+            ListNode *second_next_node = tmp->next->next;
+
+            // ------swap pointers--------
+            // tmp's next node will point to tmp
+            next_node->next = tmp;
+            // tmp's next node will be linked to its previous node
+            if(prev){
```
Comment (nitishhsinghhh) [inline Leetcode/24/Swap_Nodes_In_Pairs.cpp:39]: Use c++ way of doing things wherever applicable. 👏🏻

**Review by nitishhsinghhh — COMMENTED** [review]
Also, Please include a README.md file in the root folder for expalining your though process in few lines.

**Comment by anupamahalder** [comment]
Added README.md file a long with the updated version of the code.

**Review by nitishhsinghhh — APPROVED** [review]
LGTM

**Pull request merged**